pyrsistent/_precord.py

```diff
@@ -123,7 +126,10 @@ def set(self, key, original_value):
         if field:
             if self._factory_fields is None or field in self._factory_fields:
                 try:
-                    value = field.factory(original_value)
+                    if _PRecordEvolver._is_factory_method(field.factory):
```

Comment:
Doing inspections on live objects for each call to `set` seems like it could be potentially expensive. If you haven't already would you mind creating a small micro benchmark calling the set method repeatedly comparing the performance of tip on master with this implementation?

If there is a significant slow down I'd suggest guarding the check so that it is only executed if `self._ignore_extra` is true. Eg:
```python
if self._ignore_extra and _PRecordEvolver._is_factory_method(field.factory):
```

Reply:
@tobgu Hi Tobias, thanks your comment and review.
I've refactored code, so now no inspect needed, which was for sure overhead. Could you please check? Meanwhile I will update PClass.